src/components/WidgetGrid/index.tsx

```diff
@@ -19,15 +24,28 @@ type Props = {
   realtimeData: RealtimeDataModel;
 };
 
-function rounded(data: number | null) {
-  if (data === null) {
+function rounded(data: number | null | undefined) {
+  if (data === null || data === undefined) {
     return data;
   }
 
   return parseFloat(data.toFixed(2));
 }
 
 function WidgetGrid({realtimeData}: Props) {
+  const [aggregates, setAggregates] = useState<RideAggregate>();
+
+  useSetInterval(
+    async () => {
+      if (realtimeData.ride) {
+        const ride = await realtimeData.ride?.fetch();
+        const result = await getRideAggregates(ride);
+        setAggregates(result);
+      }
+    },
+    realtimeData.ride?.id ? 10000 : null,
```

Comment:
maybe worth making the interval time a configurable constant and put in the constants file?